packages/capabilities/src/types.ts

```diff
@@ -439,6 +442,99 @@ export interface UploadNotFound extends Ucanto.Failure {
 
 export type UploadGetFailure = UploadNotFound | Ucanto.Failure
 
+// HTTP
+export type HTTPPut = InferInvokedCapability<typeof HTTPCaps.put>
+
+// Blob
+export type Blob = InferInvokedCapability<typeof BlobCaps.blob>
+export type BlobAdd = InferInvokedCapability<typeof BlobCaps.add>
+export type ServiceBlob = InferInvokedCapability<typeof W3sBlobCaps.blob>
+export type BlobAllocate = InferInvokedCapability<typeof W3sBlobCaps.allocate>
+export type BlobAccept = InferInvokedCapability<typeof W3sBlobCaps.accept>
+
+export type BlobMultihash = Uint8Array
+export interface BlobModel {
+  digest: BlobMultihash
+  size: number
+}
+
+// Blob add
+export interface BlobAddSuccess {
+  site: {
+    'ucan/await': ['.out.ok.site', Link]
+  }
+}
+
+export interface BlobSizeOutsideOfSupportedRange extends Ucanto.Failure {
+  name: 'BlobSizeOutsideOfSupportedRange'
+}
+
+export interface AwaitError extends Ucanto.Failure {
+  name: 'AwaitError'
+}
+
+// TODO: We need Ucanto.Failure because provideAdvanced can't handle errors without it
+export type BlobAddFailure =
+  | BlobSizeOutsideOfSupportedRange
+  | AwaitError
+  | StorageGetError
+  | Ucanto.Failure
+
+export interface BlobListItem {
+  blob: BlobModel
+  insertedAt: ISO8601Date
+}
+
+// Blob allocate
+export interface BlobAllocateSuccess {
+  size: number
+  address?: BlobAddress
+}
+
+export interface BlobAddress {
+  url: ToString<URL>
+  headers: Record<string, string>
+  expiresAt: ISO8601Date
```

Comment:
We talked about not making this provider specific yesterday and the limitations that aws gives. see https://github.com/web3-storage/w3up/pull/1342#discussion_r1557906738

Reply:
I want to have a date here, otherwise I cannot infer if presigned URL is expired, and therefore allocation needs to run again. TTL is usually how long more available, rather then a date. So, we would be doing the same issue that `X-Amz-Expires` has. We could also have a receipt time to avoid this. Thoughts?

Reply:
Per wikipedia about [TTL](https://en.wikipedia.org/wiki/Time_to_live) 
> Time to live (TTL) or hop limit is a mechanism which limits the lifespan or lifetime of data in a computer or network. TTL may be implemented as a [counter](https://en.wikipedia.org/wiki/Counter_(digital)) or [timestamp](https://en.wikipedia.org/wiki/Timestamp) attached to or embedded in the data.

I mostly have seen timestamps in TTL but in datagram it is indeed more of a lifetime left, so I think you are correct. Relatedly wiki mentions [Expires](https://en.wikipedia.org/wiki/List_of_HTTP_header_fields#Responses) header and same named cookie so I think you are correct that TTL may not be a good name.


How about renaming it to `expires` just to align with HTTP header and a cookie ? I'm also ok with leaving it as is.

In regards to my comment of not replicating data already in headers, my primary motivation here is to avoid getting two out of sync, but after thinking more about it I think you are correct. We may choose to expire it sooner than what's in the presigned URL so perhaps having two out of sync is not so bad, although having our expiry higher than what's in headers still bad, but I think it's ok.